docs/tutorials/image_signing.md

```diff
@@ -0,0 +1,189 @@
+
+Let’s choose a standard `alpine` image for our signing experiment:
+
+```bash
+> sudo podman pull docker://docker.io/alpine:latest
+```
+
+```bash
+> sudo podman images alpine
+REPOSITORY                 TAG      IMAGE ID       CREATED       SIZE
+docker.io/library/alpine   latest   e7d92cdc71fe   6 weeks ago   5.86 MB
+```
+
+Now we can re-tag the image to point it to our local registry:
+
+```bash
+> sudo podman tag alpine localhost:5000/alpine
+```
+
+```bash
+> sudo podman images alpine
+REPOSITORY                 TAG      IMAGE ID       CREATED       SIZE
+localhost:5000/alpine      latest   e7d92cdc71fe   6 weeks ago   5.86 MB
+docker.io/library/alpine   latest   e7d92cdc71fe   6 weeks ago   5.86 MB
+```
+
+Podman would now be able to push the image and sign it in one command. But to
+let this work, we have to modify our system-wide registries configuration at
+`/etc/containers/registries.d/default.yaml`:
+
+```yaml
+default-docker:
+  sigstore: http://localhost:8000 # Added by us
+  sigstore-staging: file:///var/lib/containers/sigstore
+```
+
+We can see that we have two signature stores configured:
+
+- `sigstore`: referencing a web server for signature reading
+- `sigstore-staging`: referencing a file path for signature writing
+
+Now, let’s push and sign the image:
+
+```bash
+> sudo -E GNUPGHOME=$HOME/.gnupg \
+    podman push \
+    --tls-verify=false \
+    --sign-by sgrunert@suse.com \
+    localhost:5000/alpine
+…
+Storing signatures
+```
+
+If we now take a look at the systems signature storage, then we see that there
+is a new signature available, which was caused by the image push:
+
+```bash
+> sudo ls /var/lib/containers/sigstore
+'alpine@sha256=e9b65ef660a3ff91d28cc50eba84f21798a6c5c39b4dd165047db49e84ae1fb9'
+```
+
+The default signature store in
+`/etc/containers/registries.d/default.yaml` references a web server listening at
+`http://localhost:8000`. For our experiment, we simply start a new server inside
+the local staging signature store:
+
+```bash
+> sudo bash -c 'cd /var/lib/containers/sigstore && python3 -m http.server'
+Serving HTTP on 0.0.0.0 port 8000 (http://0.0.0.0:8000/) ...
+```
+
+Let’ remove the local images for our verification test:
+
+```
+> sudo podman rmi docker.io/alpine localhost:5000/alpine
+```
+
+We have to write a policy to enforce that the signature has to be valid. This
+can be done via a new rule in `/etc/containers/policy.json`:
+
+```json
+{
+  "default": [{ "type": "insecureAcceptAnything" }],
+  "transports": {
+    "docker": {
+      "localhost:5000": [
+        {
+          "type": "signedBy",
+          "keyType": "GPGKeys",
+          "keyPath": "/tmp/key.gpg"
+        }
+      ]
+    }
+  }
+}
+```
+
+The `keyPath` does not exist yet, so we have to put the GPG key there:
+
+```bash
+> gpg --output /tmp/key.pgp --armor --export sgrunert@suse.com
+```
+
+If we now pull the image:
+
+```bash
+> sudo podman pull --tls-verify=false localhost:5000/alpine
```

Comment:
This returned an error for me....  Maybe something is off in the policy.json?
```
# podman pull --tls-verify=false localhost:5000/alpine
Error: error pulling image "localhost:5000/alpine": unable to pull localhost:5000/alpine: invalid policy in "/etc/containers/policy.json": invalid character '"' after object key:value pair
```

Reply:
Guessing, that might be a missing comma; but copy&pasting the example file as written is valid JSON at least per `python -mjson.tool`.

Reply:
… and Skopeo is fine with it as well.

Reply:
Yes, @TomSweeneyRedHat can you somehow verify that your policy.json is valid?

Reply:
I still can't get the pull to work.  I created a fresh vm and installed the podman with `dnf -y install podman`.  I then added the lines for the docker section to the policy.json file and got the same error.  I wiped the policy.json file out completely and cut/paste the example in it's place.  I then was hit with an auth failure:

```
# podman pull --tls-verify=false localhost:5000/alpine
Trying to pull localhost:5000/alpine...
  Get http://localhost:8000/alpine@sha256=33b2b64e6930f7fb7710eb358429ba0d43cb6b722074952085470013a4bc6dbf/signature-1: dial tcp [::1]:8000: connect: connection refused
Error: error pulling image "localhost:5000/alpine": unable to pull localhost:5000/alpine: unable to pull image: Source image rejected: Get http://localhost:8000/alpine@sha256=33b2b64e6930f7fb7710eb358429ba0d43cb6b722074952085470013a4bc6dbf/signature-1: dial tcp [::1]:8000: connect: connection refused
```

Reply:
Yes podman is complaining that the web server (serving the signatures) is not reachable. Did you start it via the python command?

Reply:
OK two notes.  One I was not able get this to work by cut/pasting the lines into policy.json from your example.  Might be fat fingers, but kept getting the error about a quote.  However I I removed the file and replaced it with your entire example of the policy.json, then it worked.

Two, my first time through I did the python step and killed the server before moving on.  You might want to add a note like "Now in a separate window" or some such to keep dummies like me from killing the server that's needed.

Reply:
Ah I think this is the issue: https://github.com/containers/libpod/pull/5495#discussion_r395937425